### PR TITLE
[GEP-13] Add ManagedSeedSet apiserver storage

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Adds `gardener-apiserver` registration and storage for `ManagedSeedSet` resources.

Creating, updating, and deleting `ManagedSeedSet` resources can now be tested, although admission plugins / controllers are still missing. This is on purpose, to make this PR easier to review. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: hack/local-development/dev-setup-register-gardener [hunk 1]
```diff
@@ -336,6 +336,7 @@ $ADMISSION_CONTROLLER_PORT_STRING
     - UPDATE
     resources:
     - managedseeds
+    - managedseedsets
```
Comment (rfranzke) [inline hack/local-development/dev-setup-register-gardener]: I think I missed this the last time, but why are `ManagedSeed`s and `ManagedSeedSet`s part of this `validate-resource-size.gardener.cloud` webhook config? I think we only validate resources that are createable/updateable by end-users.

**Review by rfranzke — COMMENTED** [review]
/assign
Looks good, one minor question.

File: pkg/registry/seedmanagement/managedseedset/storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+}
+
+// Implement ShortNamesProvider
+var _ rest.ShortNamesProvider = &REST{}
+
+// ShortNames implements the ShortNamesProvider interface. Returns a list of short names for a resource.
+func (r *REST) ShortNames() []string {
+	return []string{}
```
Comment (timebertt) [inline pkg/registry/seedmanagement/managedseedset/storage/storage.go]: WDYT about a shortname for `ManagedSeedSet` (`mss`) and `ManagedSeed` (`ms`).
```suggestion
	return []string{"mss"}
```

File: pkg/registry/seedmanagement/managedseedset/strategy.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	return labels.Set(managedSeedSet.ObjectMeta.Labels), ToSelectableFields(managedSeedSet), nil
+}
+
+// ToSelectableFields returns a field set that represents the object.
+func ToSelectableFields(managedSeedSet *seedmanagement.ManagedSeedSet) fields.Set {
+	// The purpose of allocation with a given number of elements is to reduce
+	// amount of allocations needed to create the fields.Set. If you add any
+	// field here or the number of object-meta related fields changes, this should
+	// be adjusted.
+	shootSpecificFieldsSet := make(fields.Set, 3)
```
Comment (timebertt) [inline pkg/registry/seedmanagement/managedseedset/strategy.go]: We don't have any fields in addition to the selectable metadata fields (`namespace`, `name`).
Apart from that, you can maybe rename this var.
```suggestion
	shootSpecificFieldsSet := make(fields.Set, 2)
```

**Review by timebertt — COMMENTED** [review]
Nicely done!
Two minor things from my side.

/assign

**Comment by timebertt** [comment]
Side-note: this series of PRs has been a great review experience so far.
The individual PRs made it easy for me to review specific parts of the story in a more focused way.
Thanks for that! 👍

File: hack/local-development/dev-setup-register-gardener (hunk 1 above)
Comment (stoyanr) [reply hack/local-development/dev-setup-register-gardener]: OK, removed the entire `seedmanagement` group

File: pkg/registry/seedmanagement/managedseedset/storage/storage.go (hunk 2 above)
Comment (stoyanr) [reply pkg/registry/seedmanagement/managedseedset/storage/storage.go]: OK, added.

File: pkg/registry/seedmanagement/managedseedset/strategy.go (hunk 3 above)
Comment (stoyanr) [reply pkg/registry/seedmanagement/managedseedset/strategy.go]: OK, thanks.

**Comment by stoyanr** [comment]
@timebertt @rfranzke Thanks for your reviews, addressed your comments in a new commit.

**Review by timebertt — APPROVED** [review]
Thanks!
/lgtm

**Review by rfranzke — APPROVED** [review]

**Pull request merged**